src/pkg/packager/sources/oci.go

```diff
@@ -35,59 +35,74 @@ type OCISource struct {
 }
 
 // LoadPackage loads a package from an OCI registry.
-func (s *OCISource) LoadPackage(dst *layout.PackagePaths, unarchiveAll bool) (err error) {
+func (s *OCISource) LoadPackage(dst *layout.PackagePaths, filter filters.ComponentFilterStrategy, unarchiveAll bool) (pkg types.ZarfPackage, warnings []string, err error) {
 	ctx := context.TODO()
-	var pkg types.ZarfPackage
 	layersToPull := []ocispec.Descriptor{}
 
 	message.Debugf("Loading package from %q", s.PackageSource)
 
-	optionalComponents := helpers.StringToSlice(s.OptionalComponents)
-
 	// pull only needed layers if --confirm is set
 	if config.CommonOptions.Confirm {
-
-		layersToPull, err = s.LayersFromRequestedComponents(ctx, optionalComponents)
+		pkg, err = s.FetchZarfYAML(ctx)
+		if err != nil {
+			return pkg, nil, err
+		}
+		if filter != nil {
+			pkg.Components, err = filter.Apply(pkg)
+			if err != nil {
+				return pkg, nil, err
+			}
+		}
+		layersToPull, err = s.LayersFromRequestedComponents(ctx, pkg.Components)
 		if err != nil {
-			return fmt.Errorf("unable to get published component image layers: %s", err.Error())
+			return pkg, nil, fmt.Errorf("unable to get published component image layers: %s", err.Error())
 		}
 	}
 
 	isPartial := true
 	root, err := s.FetchRoot(ctx)
 	if err != nil {
-		return err
+		return pkg, nil, err
 	}
 	if len(root.Layers) == len(layersToPull) {
 		isPartial = false
 	}
 
 	layersFetched, err := s.PullPackage(ctx, dst.Base, config.CommonOptions.OCIConcurrency, layersToPull...)
 	if err != nil {
-		return fmt.Errorf("unable to pull the package: %w", err)
+		return pkg, nil, fmt.Errorf("unable to pull the package: %w", err)
 	}
 	dst.SetFromLayers(layersFetched)
 
-	if err := utils.ReadYaml(dst.ZarfYAML, &pkg); err != nil {
-		return err
+	// if --confirm is not set, read the zarf.yaml that was pulled
+	// and apply the filter to the components
+	if !config.CommonOptions.Confirm {
+		pkg, warnings, err = dst.ReadZarfYAML()
+		if err != nil {
+			return pkg, nil, err
+		}
+		pkg.Components, err = filter.Apply(pkg)
```

Comment:
setting pkg components like this causes issues (see above)

(clicked the wrong one)